docs/agents/GoDocs.md

```diff
@@ -5,179 +5,241 @@
+  //
+  // Returns:
+  //   - status: PENDING if still processing, FINALIZED if complete, INVALID if failed validation
+  ```
+- Expected errors documentation is mandatory (see section `Error Documentation` below)
 
-     // Example 2 - Additional context needed:
-     // GetPipeline returns the execution pipeline, or nil if not configured.
+### Error Documentation
+There are 2 categories of returned errors:
+  - Expected benign errors
+  - Exceptions (unexpected errors)
+
+Expected errors are benign sentinel errors returned by the function.
+Exceptions are unexpected errors returned by the function. These include all errors that are not benign within the context of the method.
+
+- Error classification is context-dependent - the same error type can be benign in one context but an exception in another
+- **ALL** methods that return an error **MUST** document exhaustively all expected benign errors that can be returned (if any)
+- ONLY include expected errors documentation if the function returns an error.
+- ONLY document benign errors that are expected during normal operations
+- Exceptions (unexpected errors) are NOT individually documented in the error section.
+- If sentinel errors are expected, include the catch-all statement about exceptions as the last item: `All other errors are potential indicators of bugs or corrupted internal state (continuation impossible)`
+- If no errors are expected (all return errors are exceptions), use the catch-all statement: `No errors are expected during normal operations.`
+- NEVER document individual exceptions.
+- Error documentation should be the last part of a method's documentation
+
+Before documenting any error, verify:
+- [ ] The error type exists in the codebase (for sentinel errors)
+- [ ] The error is actually returned by the method
+- [ ] The error handling matches the documented behavior
+- [ ] The error is benign in this specific context
+- [ ] If wrapping a sentinel error with fmt.Errorf, document the original sentinel error type
+- [ ] The error documentation follows the standard format
+
+Common mistakes to avoid:
+- DO NOT document errors that aren't returned
+- DO NOT document generic fmt.Errorf errors unless they wrap a sentinel error
+- DO NOT document exceptions (unexpected errors that may indicate bugs)
+- DO NOT mix benign and exceptional errors without clear distinction
+- DO NOT omit the catch-all statement about other errors
+- DO NOT document implementation details that might change
+
+#### Required Format
+
+Error documentation must follow one of these 2 formats:
+
+For methods with expected benign errors:
+```go
+// Expected error returns during normal operations:
+//   - ErrTypeName: when and why this error occurs (for sentinel errors)
+//   - ErrWrapped: when wrapped via fmt.Errorf, document the original sentinel error
+//   - All other errors are potential indicators of bugs or corrupted internal state (continuation impossible)
+```
```

Comment:
I am worried the AI will get confused, because this is presented as a go example, which is talking about wrapped errors. So this is _not_ an example for something the AI should write in a goDoc. But I also don't have a quick fix ... just ideas for details specification.